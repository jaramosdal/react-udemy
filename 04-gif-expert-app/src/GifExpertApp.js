import React, { useState } from 'react';

const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball'])

    const handleAdd = () => {
        // setCategories([...categories, 'Otra']);
        // el callback recibe el estado anterior y devuelve el nuevo
        setCategories(cats => [...cats, 'Otra']);
    }

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr />

            <button onClick={handleAdd}>Agregar</button>

            <ol>
            {
                categories.map(category => <li key={ category }>{ category }</li>)
            }
            </ol>
            
        </>
    )
};

export default GifExpertApp;