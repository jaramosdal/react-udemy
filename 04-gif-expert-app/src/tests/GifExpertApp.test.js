import React from 'react'
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {

    test('debemos mostrar <GifExpertApp /> correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debemos mostrar una lista de categorÃ­as', () => {
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });

});