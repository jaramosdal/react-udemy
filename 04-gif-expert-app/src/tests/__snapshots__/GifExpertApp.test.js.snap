// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debemos mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debemos mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</Fragment>
`;
